article/results.tex

```diff
@@ -56,59 +56,75 @@ \subsection{Best Practice Guidelines}
 As part of this work we have contributed substantial changes to the original OPFVTA repository, based on which we formulate a number of best practice guidelines, highly relevant in the production of reexecutable research outputs.
 
 \subsubsection{Errors should be fatal more often than not}
-The high complexity of a full analysis workflow makes missing output data intractable, as there are many steps at which data may have been dropped.
-Manually setting individual invocations propagate exit codes throughout the Make system is cumbersome, and thus an easy global invocation should be relied on so that errors become fatal and immediately visible.
-This is best accomplished by prepending all Bash scripts for which this concern is relevant with a \texttt{set -eu} line.
+
+By default any Python script or a compiled C or C++ program just errors out and exits immediately when running into some unexpected operation.
+POSIX shell and other similar or derived ones, such as bash and zsh,  behave differently.
+By default they continue with execution of the next scripted command, and only when the list of commands is exhausted or explicit `exit` command is called, it finally exits with a non-0 error code.
+As a result, even if not causing some detrimental actions, an execution of the script could continue for hours before it fails, and original error message might be lost in the flood of output making it hard or impossible to localize original problem to be fixed.
+Analogously, shell scripts treat undefined variables as a variable having an empty value, and thus not causing any error at all.
+As a result a command in a script like \texttt{rm -rf "\${PREFIX}/"} would do its best to try to remove everything on your harddrive in case \texttt{PREFIX} variable is not defined.
+\texttt{set -u} would change the default behavior to error out and stop execution as soon as undefined variable is attempted to be used.
```

Comment:
Here is from the comment above (but with the last line removed) in a correct "multiline" suggestion selected mode:

```suggestion
As a result, even if not causing some detrimental actions, an execution of the script could continue for hours before it fails, and original error message might be lost in the flood of output making it hard or impossible to localize original problem to be fixed.
This can be addressed by prepending \texttt{set -e} to the respective shell script, which changes the default behavior so that it will error out and stop execution as command exits with an error code.
Additionally, shell scripts treat undefined variables as a variable having an empty value, with empty values causing any error at all.
This can lead to numerous ill-defined behaviours, including a command such as \texttt{rm -rf "\${PREFIX}/"} removing all files on the system if \texttt{PREFIX} is not defined.
This can be addressed by prepending \texttt{set -u} so that an error is raised and execution is stopped as soon as undefined variable is attempted to be used.
```